internal/provisioner/controller/gatewayclass.go

```diff
@@ -194,6 +194,12 @@ func (r *gatewayClassReconciler) Reconcile(ctx context.Context, req ctrl.Request
 					}
 				}
 			}
+
+			if params.Spec.Envoy.LogLevel != "" && params.Spec.Envoy.LogLevel.EnvoyValidate() != nil {
```

Comment:
could this be simplified to not have the empty string check? if `EnvoyValidate` returns `nil` when log level is empty (could do the same for contour log level but i dont see that method called anywhere)

Reply:
also tbh @skriss' comments probably trump this so can revisit once his requested changes are made